### PR TITLE
Missing context new status for CA

### PR DESCRIPTION
**Pull request by barv-jfrog**
- [x] The pull request is targeting the `dev` branch.
- [x] The code has been validated to compile successfully by running `go vet ./...`.
- [x] The code has been formatted properly using `go fmt ./...`.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-security/actions/workflows/analysis.yml) passed.
- [x] All [tests](https://github.com/jfrog/jfrog-cli-security/actions/workflows/test.yml) have passed. If this feature is not already covered by the tests, new tests have been added.
- [x] All changes are detailed at the description. if not already covered at [JFrog Documentation](https://github.com/jfrog/documentation), new documentation have been added.

-----
Description:
New status in contextual analysis - missing context. If a CVE gets a missing context under contextual analysis this means that this CVE exists on binary but not on source code. (For example a scanner exists but is of type ELF, YAML, etc - not source code). This helps to distinguish between Not covered which means this CVE doesn't have a jfrog scanner at all to Missing context which means the CVE is covered but not on source code.

Since missing context doesn't mean Not applicable it has a higher priority. Also its not that its not Covered, just not covered in source code, which in conclusion we believe deserves a higher priority than Not covered as well.

Additional description:
Jfrog research team work on many CVEs using scanners they create. These scanners have a type, like JS, JAVASRC, YAML, etc..
When using jf audit we scan for src code, and therefore the scanners which can be run on the relevant project are only scanners of source code types which are being supported by the CLI (JAVASRC, JS, Python, NuGet soon and more) but not on scanners that are of type YAML,  text, binary, elf, etc, however, these CVEs are supported on binary scanning on xray platform - So we want to distinguish between those CVEs that are not covered at all (no scanner) and those that are covered on xray platform but not on source code scanning (jf audit).

This CA status is only for jf audit.

### CODE REVIEW

**Comment by hadarshjfrog** [comment]
Please add relevant results to SARIFs in: `tests/testdata/other/applicability-scan`.
Also - can you add a project that will return such results when being scan?

File: utils/severityutils/severity.go [hunk 1]
```diff
@@ -88,32 +88,37 @@ func (sd SeverityDetails) ToDetails(severity Severity, pretty bool) formats.Seve
 
 var Severities = map[Severity]map[jasutils.ApplicabilityStatus]*SeverityDetails{
 	Critical: {
-		jasutils.Applicable:                &SeverityDetails{Priority: 20, Score: MaxCveScore, Emoji: "💀", style: color.New(color.BgLightRed, color.LightWhite)},
-		jasutils.ApplicabilityUndetermined: &SeverityDetails{Priority: 19, Score: MaxCveScore, Emoji: "💀", style: color.New(color.BgLightRed, color.LightWhite)},
+		jasutils.Applicable:                &SeverityDetails{Priority: 21, Score: MaxCveScore, Emoji: "💀", style: color.New(color.BgLightRed, color.LightWhite)},
+		jasutils.ApplicabilityUndetermined: &SeverityDetails{Priority: 20, Score: MaxCveScore, Emoji: "💀", style: color.New(color.BgLightRed, color.LightWhite)},
+		jasutils.MissingContext:            &SeverityDetails{Priority: 19, Score: MaxCveScore, Emoji: "💀", style: color.New(color.BgLightRed, color.LightWhite)},
```
Comment (attiasas) [inline utils/severityutils/severity.go:93]: you have entries with similar Priority values. make sure that you don't have duplications in values.

File: utils/resultstable.go [hunk 2]
```diff
@@ -1057,6 +1060,9 @@ func getFinalApplicabilityStatus(applicabilityStatuses []jasutils.ApplicabilityS
 	foundUndetermined := false
 	foundNotCovered := false
 	for _, status := range applicabilityStatuses {
+		if status == jasutils.MissingContext {
+			return jasutils.MissingContext
+		}
```
Comment (attiasas) [inline utils/resultstable.go:1065]: Missing Context does not have the first priority.
If we have a case of cve with MissingContext and another Cve with Applicable. we want to return the Applicable. only if not Applicable you can return the `MissingContext`, make it similar logic to `ApplicabilityUndetermined`

File: utils/jasutils/jasutils.go [hunk 3]
```diff
@@ -75,9 +78,10 @@ func ApplicabilityRuleIdToCve(sarifRuleId string) string {
 }
 
 var applicableMapToScore = map[string]int{
-	"Applicable":                4,
-	"ApplicabilityUndetermined": 3,
-	"NotScanned":                2,
+	"Applicable":                5,
+	"ApplicabilityUndetermined": 4,
+	"NotScanned":                3,
+	"MissingContext":            2,
```
Comment (attiasas) [inline utils/jasutils/jasutils.go:84]: make sure priority is matching the order in all places

**Review by attiasas — CHANGES_REQUESTED** [review]
Check out my comments. Also. add to integration tests the status. (validation + issue with the status)
Like in the validation methods: `VerifySimpleJsonJasResults`

File: utils/severityutils/severity.go (hunk 1 above)
Comment (barv-jfrog) [reply utils/severityutils/severity.go:93]: no duplications

File: utils/jasutils/jasutils.go (hunk 3 above)
Comment (barv-jfrog) [reply utils/jasutils/jasutils.go:84]: made sure

**Review by attiasas — APPROVED** [review]

File: tests/testdata/projects/package-managers/go/missing-context/go.mod [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+go 1.22
+
+require (
+	cloud.google.com/go v0.38.0 // indirect
+	github.com/Azure/azure-sdk-for-go v40.3.0+incompatible // indirect
+	github.com/Azure/go-autorest/autorest v0.10.0 // indirect
+	github.com/Azure/go-autorest/autorest/adal v0.8.2 // indirect
+	github.com/Azure/go-autorest/autorest/azure/auth v0.4.2 // indirect
+	github.com/Azure/go-autorest/autorest/azure/cli v0.3.1 // indirect
```
Comment (attiasas) [inline tests/testdata/projects/package-managers/go/missing-context/go.mod:11]: Please only include the needed dependency for the test to reduce size and test times

**Review by attiasas — CHANGES_REQUESTED** [review]

**Review by attiasas — APPROVED** [review]

**Pull request merged**